pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,323 @@
+package serviceaccount
+
+import (
+	"fmt"
+
+	"github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"strings"
+
+	v13 "github.com/rancher/types/apis/core/v1"
+	v12 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	v1beta12 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const apiGroup = "rbac.authorization.k8s.io"
+const apiVersion = "extensions/v1beta1"
+
+const podSecurityAnnotation = "serviceaccount.cluster.cattle.io/pod-security"
+const projectIdAnnotation = "field.cattle.io/projectId"
+
+func Register(context *config.UserContext) {
+	logrus.Infof("registering serviceaccount handler for cluster %v", context.ClusterName)
+
+	m := &manager{
+		clusters: context.Management.Management.Clusters(""),
+		bindings: context.RBAC.RoleBindings(""),
+		policies: context.Extensions.PodSecurityPolicies(""),
+		roles:    context.RBAC.Roles(""),
+
+		clusterLister:   context.Management.Management.Clusters("").Controller().Lister(),
+		templateLister:  context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		policyLister:    context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		bindingLister:   context.RBAC.RoleBindings("").Controller().Lister(),
+		roleLister:      context.RBAC.Roles("").Controller().Lister(),
+		namespaceLister: context.Core.Namespaces("").Controller().Lister(),
+		projectLister:   context.Management.Management.Projects("").Controller().Lister(),
+	}
+
+	/*
+		var mutex = &sync.Mutex{}
+		shouldBindPspt := true
+	*/
+
+	context.Core.ServiceAccounts("").AddHandler(
+		"universalServiceAccountHandler",
+		func(key string, obj *v1.ServiceAccount) error {
+			/*
+				r := func() {
+					mutex.Lock()
+					defer mutex.Unlock()
+					if shouldBindPspt {
+						shouldBindPspt = false
+
+						google11, err := m.clusterLister.Get("", "local")
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+						google11.Spec.DefaultPodSecurityPolicyTemplateName = "pspt1000"
+
+						_, err = m.clusters.Update(google11)
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+					}
+				}
+
+				if true {
+					r()
+				}
+			*/
+			logrus.Info("handle service account")
+
+			if obj == nil {
+				logrus.Infof("no service account provided, exiting")
+				// do nothing
+				return nil
+			}
+
+			namespaceName := obj.Namespace
+
+			logrus.Infof("svc acct name: %v namespace: %v", obj.Name, namespaceName)
+
+			// Check if a PSP exists for this cluster
+			policies, err := m.policyLister.List("", labels.Everything())
+
```

Comment:
This is very small stylistic nitpick but the common style ive seen is for there to be no empty line between the return of an error and checking if its nil.
so, do this:
```
policies, err := m.policyLister.List("", labels.Everything())
if err != nil {
...
}
```
not this:
```
policies, err := m.policyLister.List("", labels.Everything())

if err != nil {
...
}
```